### PR TITLE
Updated max_key_fee_setting

### PR DESCRIPTION
**Pull request by hackrush01**
Now setting max_key_fee to 0 will mean that max_key_fee is unlimited.

### CODE REVIEW

**Comment by hackrush01** [comment]
Goes with this PR [lbry-app/#324](https://github.com/lbryio/lbry-app/pull/324)

**Comment by kaykurokawa** [comment]
User might actually want to set it to zero though ( if he only wants free content).
Setting it to negative is an option. 
Personally I'd prefer something less hacky like a new setting variable like disable_max_key_fee_check.

**Comment by kauffj** [comment]
@kaykurokawa it looks like @hackrush01 updated this to work as requested and use an additional disable flag. Is this mergeable?

File: lbrynet/daemon/Downloader.py [hunk 1]
```diff
@@ -97,7 +98,7 @@ def check_fee_and_convert(self, fee):
                                                                            fee.amount)
         if converted_fee_amount > self.wallet.get_balance():
             raise InsufficientFundsError('Unable to pay the key fee of %s' % converted_fee_amount)
-        if converted_fee_amount > max_key_fee_amount:
+        if converted_fee_amount > max_key_fee_amount and not disable_max_key_fee:
```
Comment (kaykurokawa) [inline lbrynet/daemon/Downloader.py]: missing self. here

**Comment by kaykurokawa** [comment]
I will recommend that you write unit test for this function https://github.com/lbryio/lbry/blob/master/lbrynet/daemon/Downloader.py#L94 to make sure your max key fee and disable key fee settings are working. 

Here is where the unit tests for the Downloader.py file goes: 
https://github.com/lbryio/lbry/blob/master/tests/unit/lbrynet_daemon/test_Downloader.py

You can run the twisted trial tests for lbry by running PYTHONPATH=. trial tests in the base folder ( or to run specific tests you can run something like PYTHONPATH=. trial tests/unit/lbrynet_daemon/test_Something.py )

**Comment by hackrush01** [comment]
Ok, I'll look into it.

**Comment by hackrush01** [comment]
@kaykurokawa I DM'ed you on slack, regarding this issue.

**Comment by kaykurokawa** [comment]
I tested the disable_max_key_fee setting and works as expected. 
Let's remove the new unit tests you attempted for now (below line 186 on test_Downloader.py)  and merge this.
 
And then we can open up a new PR to finish writing the tests which I can help you out on.

**Comment by hackrush01** [comment]
Ok, yeah. That would be great, I'll just make the changes and push it.

**Comment by hackrush01** [comment]
@kaykurokawa Please check.

**Comment by kaykurokawa** [comment]
will be merged from here https://github.com/lbryio/lbry/pull/781 , thanks

**Pull request closed without merging**